native/c/makefile

```diff
@@ -56,12 +56,13 @@ C_FLAGS=-W "c,lp64,langlvl(extended),xplink,exportall"\
 
 CXXLANG_FLAGS=-W "c,lp64"\
  -c \
- -I./chdsect
+ -I./chdsect \
+ -DNO_SHMEM
 
 SWIG_FLAGS=-W "c,lp64,define(SWIG)"\
  -c \
  -I./chdsect \
- -DSWIG
+ -DSWIG -DNO_SHMEM
```

Comment:
Usually its best to gate these kinds of features with macros so that we only use it when its enabled. This implementation implies the opposite, where the developer has to add this build flag to prevent the shared memory from being used. (positive vs. negative macros)

Can we refactor this so that we pass a flag to enable the feature in our build, and omit this flag from extender builds?

For reference, this is what we do with ZLOG_ENABLE and SWIG - these flags need passed to enable the feature, whereas here we need to pass the NO_SHMEM macro to disable it. Also, I wonder if this would be a good opportunity to establish some consistency with the macros - we could rename to ZSHMEM_ENABLE or something along those lines